src/sagemaker/cli/compatibility/v2/modifiers/predictors.py

```diff
@@ -0,0 +1,146 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Classes to modify Predictor code to be compatible
+with version 2.0 and later of the SageMaker Python SDK.
+"""
+from __future__ import absolute_import
+
+import ast
+
+from sagemaker.cli.compatibility.v2.modifiers.modifier import Modifier
+
+BASE_PREDICTOR = "RealTimePredictor"
+PREDICTORS = {
+    "FactorizationMachinesPredictor": ("sagemaker", "sagemaker.amazon.factorization_machines"),
+    "IPInsightsPredictor": ("sagemaker", "sagemaker.amazon.ipinsights"),
+    "KMeansPredictor": ("sagemaker", "sagemaker.amazon.kmeans"),
+    "KNNPredictor": ("sagemaker", "sagemaker.amazon.knn"),
+    "LDAPredictor": ("sagemaker", "sagemaker.amazon.lda"),
+    "LinearLearnerPredictor": ("sagemaker", "sagemaker.amazon.linear_learner"),
+    "NTMPredictor": ("sagemaker", "sagemaker.amazon.ntm"),
+    "PCAPredictor": ("sagemaker", "sagemaker.amazon.pca"),
+    "RandomCutForestPredictor": ("sagemaker", "sagemaker.amazon.randomcutforest"),
+    "RealTimePredictor": ("sagemaker", "sagemaker.predictor"),
+    "SparkMLPredictor": ("sagemaker.sparkml", "sagemaker.sparkml.model"),
+}
+
+
+class PredictorConstructorRefactor(Modifier):
+    """A class to refactor *Predictor class and refactor endpoint attribute."""
+
+    def node_should_be_modified(self, node):
+        """Checks if the ``ast.Call`` node instantiates a class of interest.
+
+        This looks for the following calls:
+
+        - ``sagemaker.<my>.<namespace>.<MyPredictor>``
+        - ``sagemaker.<namespace>.<MyPredictor>``
+        - ``<MyPredictor>``
+
+        Args:
+            node (ast.Call): a node that represents a function call. For more,
+                see https://docs.python.org/3/library/ast.html#abstract-grammar.
+
+        Returns:
+            bool: If the ``ast.Call`` instantiates a class of interest.
+        """
+        return any(_matching(node, name, namespaces) for name, namespaces in PREDICTORS.items())
+
+    def modify_node(self, node):
+        """Modifies the ``ast.Call`` node to call ``Predictor`` instead.
+
+        Also renames ``endpoint`` attribute to ``endpoint_name``.
+
+        Args:
+            node (ast.Call): a node that represents a *Predictor constructor.
+        """
+        _rename_class(node)
+        _rename_endpoint(node)
+
+
+def _matching(node, name, namespaces):
```

Comment:
[out of scope] I think we might be getting to the point where it makes sense to refactor these types of methods into a common module

Reply:
i would agree. was thinking the same thing. i thought i'd get these methods out here to think about.